shared/js/l10n.js

```diff
@@ -1607,27 +1684,32 @@
       }
     }
 
-    if (!this.ctx.availableLocales.length) {
-      // if there was no availableLanguages meta,
-      // register the default locale only
-      this.ctx.registerLocales(this.ctx.defaultLocale);
+    if (navigator.mozApps && navigator.mozApps.getAdditionalLanguages) {
+      navigator.mozApps.getAdditionalLanguages().then(
+        registerLocales.bind(this, meta));
+    } else {
+      registerLocales.call(this, meta);
     }
-    return initLocale.call(this);
```

Comment:
Unfortunately that pushes us way beyond firstPaint because of the amount of async your code causes (4 instead of 1).

Even if that was not the case, I believe that your code would still be slower because of the amount of promises and since it is a hot-path in runtime, I'd be against it.

Even if that was not the case, your concept of what is easy to understand somehow ignores pushing for Function.prototype.apply.bind monsters which I'd like to avoid at all cost.